src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+    // If not, ask the delegate if it's ok.  If yes, add to our set and proceed, if not, fail.
+    //
+    if (!KnownVideoStreamID(videoStreamID))
+    {
+        // Call the delegate to validate that the videoStreamID is known; if yes then add to our list and proceed
+        //
+        if (!mDelegate->IsValidVideoStreamID(videoStreamID))
+        {
+            ChipLogError(Zcl, "Unknown Video Stream ID provided.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+        AddMoveCapableVideoStreamID(videoStreamID);
+        MarkDirty(Attributes::DPTZRelativeMove::Id);
+    }
+
+    // Call the delegate
+    Status status = mDelegate->DPTZSetViewport(videoStreamID, viewport);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleDPTZRelativeMove(HandlerContext & ctx,
+                                                                 const Commands::DPTZRelativeMove::DecodableType & commandData)
+{
+    uint16_t videoStreamID     = commandData.videoStreamID;
+    Optional<int16_t> deltaX   = commandData.deltaX;
+    Optional<int16_t> deltaY   = commandData.deltaY;
+    Optional<int8_t> zoomDelta = commandData.zoomDelta;
+
+    // Verify that a received Zoom Delta is within constraints
+    //
+    if (zoomDelta.HasValue())
+    {
+        int8_t zoomDeltaValue = zoomDelta.Value();
+        if (zoomDeltaValue < -100 || zoomDeltaValue > 100)
+        {
+            ChipLogError(Zcl, "Provided Digital Zoom Delta is out of range");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+    }
+    // Is this a video stream ID of which we have already been informed?
+    // If not, ask the delegate if it's ok.  If yes, add to our set and proceed, if not, fail.
+    //
+    if (!KnownVideoStreamID(videoStreamID))
+    {
+        // Call the delegate to validate that the videoStreamID is known; if yes then add to our list and proceed
+        //
+        if (!mDelegate->IsValidVideoStreamID(videoStreamID))
+        {
+            ChipLogError(Zcl, "Unknown Video Stream ID provided.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+        AddMoveCapableVideoStreamID(videoStreamID);
+        MarkDirty(Attributes::DPTZRelativeMove::Id);
+    }
+
+    // Call the delegate
+    Status status = mDelegate->DPTZRelativeMove(videoStreamID, deltaX, deltaY, zoomDelta);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
```

Comment:
This seems like a complicated way to write:
```suggestion
    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
```

Reply:
Fixed